src/server/room-controller.js

```diff
@@ -195,6 +214,12 @@ export default class RoomController {
       (req, res) => this.handlePatchRoom(req, res)
     );
 
+    router.delete(
+      '/api/v1/rooms',
+      [needsPermission(permissions.OWN_ROOMS), validateQuery(deleteRoomsQuerySchema)],
```

Comment:
I think we need to use another permission here, one that is only allocated to admins (`OWN_ROOMS` is a permission normal users have as well)

Reply:
But I was under the impression that every user should be able to delete their own rooms?

Reply:
But they do that room by room, right? That's basically the `DELETE /api/v1/rooms/:roomId` endpoint.
I thought this endpoint is for an admin to delete all private rooms of a user (since we pass "ownerId"). For cases where users should delete their own rooms, we should check that the `req.user` is the owner.

Reply:
Done, I also renamed the handler to `handleDeleteAllRoomsForUser` in order to clearly distinguish it from the "normal" delete handler.